nanoFramework.WebServer/HttpMultipartParser/MultipartFormDataParser.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.IO;
+using System.Text;
+
+namespace nanoFramework.WebServer.HttpMultipartParser
+{
+    /// <summary>
+    ///     Provides methods to parse a
+    ///     <see href="http://www.ietf.org/rfc/rfc2388.txt">
+    ///         <c>multipart/form-data</c>
+    ///     </see>
+    ///     stream into it's parameters and file data.
+    /// </summary>
+    public class MultipartFormDataParser
+	{
+        private const int defaultBufferSize = 4096;
+
+        private readonly bool _ignoreInvalidParts;
+        private readonly int _binaryBufferSize;
+        private string _boundary;
+        private byte[] _boundaryBinary;
+        private readonly Stream _stream;
+        private bool _readEndBoundary;
+        private readonly ArrayList _files = new();
+        private readonly ArrayList _parameters = new();
+
+        /// <summary>Initializes a new instance of the <see cref="MultipartFormDataParser" /> class</summary>
+		/// <param name="stream">The stream containing the multipart data.</param>
```

Comment:
quite some weird indent in the file. 'ctrl + k + d' should fix all those

Reply:
Weird indeed, tabs before comments were not replaced by spaces by ctrl+k+d. Had to force it manually. Fixed.